transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -89,7 +91,8 @@ public SimpleChannelPool(Bootstrap bootstrap, final ChannelPoolHandler handler,
      */
     public SimpleChannelPool(Bootstrap bootstrap, final ChannelPoolHandler handler, ChannelHealthChecker healthCheck,
                              boolean releaseHealthCheck) {
-        this.handler = checkNotNull(handler, "handler");
+        ChannelPoolStats statsHandler = new ChannelPoolStats(checkNotNull(handler, "handler"));
```

Comment:
If all the channel pooling  will be rewritten, It makes sense to add the relevant stats there,
and ditch this one.